### ISSUE TITLE
Incomplete support for PCSC reader

### ISSUE THREAD

**Issue by openmilanese** [bas-t/descrambler#13]
I try to compile the descrambler with PCSC enable but not found
From  descrambler/examples/cardslot.conf.example:
; For using this cardslot type, you must have installed pcsd-lite and ccid
; driver package. Before compiling you should add something like this to your
; Make.config file:
; WITH_PCSC = 1
; INCLUDES += -I/usr/include/PCSC

I also install libpcsclite-dev for create the include /usr/include/PCSC and i add the directive in LIBS:
##################################
VERSION = $(shell git describe --always)

include config.mak

CC       ?= gcc
CXX      ?= g++
CXXFLAGS ?= -Wall -D__user= 

DEFINES += -DRELEASE_VERSION=\"$(VERSION)\" -D__KERNEL_STRICT_NAMES
LBDIR = src
TOOL = ffdecsawrapper
LIBS = -lcrypto -lcrypt -lv4l1 -lpthread **-lpcsclite**
SCDIR = sc/PLUGINS/src
SCLIBS = -Wl,-whole-archive ./sc/PLUGINS/lib/libsc-*.a -Wl,-no-whole-archive \
	./sc/PLUGINS/lib/libffdecsawrapper-sc.a

**WITH_PCSC = 1
INCLUDES += -I/usr/include/PCSC**

ifneq ($(RELEASE),1)
CXXFLAGS += -g
endif
......
......
......
########################################
This parameter is on main Makefile  descrambler/Makefile

Total software installed is:
ii  libccid                              1.4.22-1ubuntu0.1                                amd64        PC/SC driver for USB CCID smart card readers
ii  libpcsc-perl                         1.4.14-1build1                                   amd64        Perl interface to the PC/SC smart card library
ii  libpcsclite-dev                      1.8.14-1ubuntu1.16.04.1                          amd64        Middleware to access a smart card using PC/SC (development files)
ii  libpcsclite1:amd64                   1.8.14-1ubuntu1.16.04.1                          amd64        Middleware to access a smart card using PC/SC (library)
ii  pcsc-tools                           1.4.25-1                                         amd64        Some tools to use with smart cards and PC/SC
ii  pcscd                                1.8.14-1ubuntu1.16.04.1                          amd64        Middleware to access a smart card using PC/SC (daemon side)

Compile is always ok whit this parameter but ffdecsawrapper not recognized the ccid reader:
```
loading cardslot config from /etc/camfiles/cardslot.conf
unknown cardslot type 'ccid'
```

**Comment by bas-t** [comment]
ffdecsawrapper does not communicate directly with your smartcard. Oscam does.
So setup oscam to do it's job and setup ffdecsawrapper to communicate with oscam.

**Comment by openmilanese** [comment]
I have been using dvbloopback + sasc-ng (now ffdecsawrapper)> newcamd> oscam + pcsc in combination with mythtv for many years now and I can give you a good answer to how the solution works.
The former sasc-ng, however, has in fact always had direct operation with the reading of the smartcard, and the file, for example, in the source cardslot.conf is proof of this.
Is my analysis correct?
In the course of the software remake, did you only consolidate the operation with transport protocols (newcamd, cccam, etc.)?
If so, it would be worthwhile to delete the sample file cardslot.conf ...

**Comment by bas-t** [comment]
I guess you are right, about the cardslot support.
However, I'm not using this software anymore. I shifted to tvheadend. Tvheadend has built-in ffdecsa,
So, if you want to improve this little piece of software, just send me a pull request.

**Comment by openmilanese** [comment]
Hello bas-t, I restored pcsc reader support. Unfortunately the part concerning the cam nagravision (and perhaps also the other encodings) is old-fashioned compared to oscam, for example the tigerkey does not exist at 240 characters ....
It is therefore unusable.
I have also experienced tvheadend these days and we are indeed on another planet as decoding. Fast, stable and easy to use.
I redesign my HTPC with Tvheadend as PVR, Emby as a media server and Kodi as a frontend.
I was very fond of mythtv but it was time for a change.
However, I send the PCSC patch series to be integrated into your code.

-Dependence:
`apt-get install libpcsclite-dev libccid pcscd pcsc-tools`

-Content of configuration file cardslot.conf (read output of the command "pcsc_scan"):
`ccid:OMNIKEY AG CardMan 3121 00 00`

-Log outuput ok:
```
Apr 20 10:00:11.397 CAM(general.info): loading cardslot config from /etc/camfiles/cardslot.conf
Apr 20 10:00:11.972 CAM(core.smartcard): 0: establish context: 0:Command successful.
Apr 20 10:00:11.977 CAM(core.smartcard): 0: connect 'OMNIKEY AG CardMan 3121 00 00': 0:Command successful.
```

-Patch:
[Makefile-master.patch](https://github.com/bas-t/descrambler/files/1937147/Makefile-master.patch.txt)
[Makefile-sc.patch](https://github.com/bas-t/descrambler/files/1937148/Makefile-sc.patch.txt)
